File: python/003/main.py
# ProjectEuler 003
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`


def main():
    n = 600851475143
    p = 2
    while n > 1:
        if n % p == 0:
            n /= p
        else:
            p += 1
    print(p)

main()

File: julia/003/main.jl
# ProjectEuler 003
# Julia / main.jl
# copyright (c) 2015 Susisu
#
# Julia v0.4.1
# `julia main.jl`

function main()
    n = 600851475143
    d = 2
    while n > 1
        if n % d == 0
            n /= d
        else
            d += 1
        end
    end
    println(d)
end

main()
